### DISCUSSION TITLE
Using beforeRequest to alter the body content. Content-length invalid based on documentation example.

### DISCUSSION THREAD

**Discussion by simplecommerce** [sindresorhus/got#2197] [General]
Hi,

I just wanted to mention that in my existing project, I tried to follow the following example and it didn't work for me in my case.

https://github.com/sindresorhus/got/blob/main/documentation/9-hooks.md#beforerequest

```
import got from 'got';

const response = await got.post(
	'https://httpbin.org/anything',
	{
		json: {payload: 'old'},
		hooks: {
			beforeRequest: [
				options => {
					options.body = JSON.stringify({payload: 'new'});
					options.headers['content-length'] = options.body.length.toString();
				}
			]
		}
	}
);
```

In my case, when I added my key directly in the json body and logged the options being sent, the content-length differed from doing it in the method explained above.

I had to use `options.headers["content-length"] = Buffer.byteLength(options.body)` instead of `options.body.length.toString()` to make it work.

Just felt like I should mention it in case other people have the same issue.

**Comment by sindresorhus** [comment]
Fixed: https://github.com/sindresorhus/got/commit/5bdd3d5f175c5e0d8e37dbb7e8a946171390de17
